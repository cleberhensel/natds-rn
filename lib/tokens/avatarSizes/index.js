import { sizes } from '../sizes';

export const avatarSizes = {
  tiny: {
    size: sizes.standard,
    fontSize: '.625rem',
  },
  small: {
    size: sizes.semi,
    fontSize: '.875rem',
  },
  standard: {
    size: sizes.semix,
    fontSize: '1rem',
  },
  large: {
    size: sizes.medium,
    fontSize: '1rem',
  },
  huge: {
    size: sizes.largexx,
    fontSize: '1.5rem',
  },
};
