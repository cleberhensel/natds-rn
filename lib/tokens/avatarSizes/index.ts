import { sizes } from "../sizes";
export const avatarSizes = {
  huge: {
    fontSize: "1.5rem",
    size: sizes.largexx
  },
  large: {
    fontSize: "1rem",
    size: sizes.medium
  },
  small: {
    fontSize: ".875rem",
    size: sizes.semi
  },
  standard: {
    fontSize: "1rem",
    size: sizes.semix
  },
  tiny: {
    fontSize: ".625rem",
    size: sizes.standard
  }
};
