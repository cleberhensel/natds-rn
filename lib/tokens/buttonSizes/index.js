import { sizes } from '../sizes';

export const buttonSizes = {
  small: {
    paddingTop: sizes.tiny,
    paddingRight: sizes.tiny,
    paddingBottom: sizes.tiny,
    paddingLeft: sizes.tiny,
    minHeight: sizes.semi,
  },
  medium: {
    paddingTop: 12,
    paddingRight: sizes.small,
    paddingBottom: 12,
    paddingLeft: sizes.small,
    minHeight: sizes.semix,
  },
  large: {
    paddingTop: sizes.small,
    paddingRight: sizes.small,
    paddingBottom: sizes.small,
    paddingLeft: sizes.small,
    minHeight: sizes.medium,
  },
};
