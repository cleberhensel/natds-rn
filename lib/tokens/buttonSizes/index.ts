import { sizes } from "../sizes";
export const buttonSizes = {
  large: {
    minHeight: sizes.medium,
    paddingBottom: sizes.small,
    paddingLeft: sizes.small,
    paddingRight: sizes.small,
    paddingTop: sizes.small
  },
  medium: {
    minHeight: sizes.semix,
    paddingBottom: 12,
    paddingLeft: sizes.small,
    paddingRight: sizes.small,
    paddingTop: 12
  },
  small: {
    minHeight: sizes.semi,
    paddingBottom: sizes.tiny,
    paddingLeft: sizes.tiny,
    paddingRight: sizes.tiny,
    paddingTop: sizes.tiny
  }
};
