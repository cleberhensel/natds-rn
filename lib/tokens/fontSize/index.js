export const fontSize = {
  h1: {
    fontSize: 96,
    fontWeight: 400,
    lineHeight: 144,
  },
  h2: {
    fontSize: 60,
    fontWeight: 400,
    lineHeight: 90,
  },
  h3: {
    fontSize: 48,
    fontWeight: 400,
    lineHeight: 72,
  },
  h4: {
    fontSize: 34,
    fontWeight: 400,
    lineHeight: 51,
  },
  h5: {
    fontSize: 24,
    fontWeight: 400,
    lineHeight: 36,
  },
  h6: {
    fontSize: 20,
    fontWeight: 500,
    lineHeight: 30,
  },
  subtitle1: {
    fontSize: 16,
    fontWeight: 500,
    lineHeight: 24,
  },
  subtitle2: {
    fontSize: 14,
    fontWeight: 500,
    lineHeight: 21,
  },
  body1: {
    fontSize: 16,
    fontWeight: 400,
    lineHeight: 24,
  },
  body2: {
    fontSize: 14,
    fontWeight: 400,
    lineHeight: 21,
  },
  button: {
    fontSize: 14,
    fontWeight: 500,
    lineHeight: 21,
  },
  caption: {
    fontSize: 12,
    fontWeight: 400,
    lineHeight: 18,
  },
  overline: {
    fontSize: 12,
    fontWeight: 500,
    lineHeight: 18,
  },
};
