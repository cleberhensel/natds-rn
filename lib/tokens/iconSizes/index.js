import { sizes } from '../sizes';

export const iconSizes = {
  micro: sizes.small,
  tiny: sizes.standard,
  small: sizes.semi,
  standard: sizes.medium,
  medium: sizes.large,
  large: sizes.largex,
  largex: sizes.huge,
  largexx: sizes.hugex,
  huge: sizes.hugexx,
  hugex: sizes.hugexxx,
  hugexx: sizes.veryhuge,
};
