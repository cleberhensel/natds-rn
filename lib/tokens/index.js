import { colors as colorNatura } from './colors/natura';

export { avatarSizes } from './avatarSizes';
export { buttonSizes } from './buttonSizes';
export { elevation } from './elevation';
export { iconSizes } from './iconSizes';
export { opacity } from './opacity';
export { overlay } from './overlay';
export { spacing } from './spacing';
export { radius } from './radius';
export { fontSize } from './fontSize';

const colors = {
  natura: colorNatura,
};

export { colors };
