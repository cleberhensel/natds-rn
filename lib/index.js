import * as Tokens from './tokens';
import { ThemeNaturaLight } from './themes/ThemeNaturaLight';

export { NatContainer } from './components/NatContainer';
export { NatHeader } from './components/NatHeader';
export { NatButton } from './components/NatButton';
export { NatGridItem } from './components/NatGridItem';
export { NatModal } from './components/NatModal';
export { NatLoading } from './components/NatLoading';
export { NatCard } from './components/NatCard';
export { NatTextInput } from './components/NatTextInput';
export { NatLogoVertical } from './components/NatLogoVertical';
export { NatLogoHorizontal } from './components/NatLogoHorizontal';
export { NatMenuItem } from './components/NatMenuItem';
export { NatImage } from './components/NatImage';
export { NatIcon } from './components/NatIcon';
export { NatLogo } from './components/NatLogo';
export { NatText } from './components/NatText';
export { NatLink } from './components/NatLink';
export { Icon } from './assets/icons';
export { NatCounterToggler } from './components/NatCounterToggler';
export { NatPeriodButton } from './components/NatPeriodButton';
export { NatGenericCard } from './components/NatGenericCard';
export { NatSpace } from './components/NatSpace';
export { Tokens };
export { ThemeNaturaLight };
