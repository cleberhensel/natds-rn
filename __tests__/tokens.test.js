import * as tokens from '../lib/tokens';

describe('Tokens', () => {
  test('It should match tokens snapshot', () => {
    expect(tokens).toMatchSnapshot('Tokens Snapshot');
  });
});
