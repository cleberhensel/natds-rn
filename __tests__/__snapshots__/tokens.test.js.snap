// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tokens It should match tokens snapshot: Tokens Snapshot 1`] = `
Object {
  "avatarSizes": Object {
    "huge": Object {
      "fontSize": "1.5rem",
      "size": 80,
    },
    "large": Object {
      "fontSize": "1rem",
      "size": 48,
    },
    "small": Object {
      "fontSize": ".875rem",
      "size": 32,
    },
    "standard": Object {
      "fontSize": "1rem",
      "size": 40,
    },
    "tiny": Object {
      "fontSize": ".625rem",
      "size": 24,
    },
  },
  "buttonSizes": Object {
    "large": Object {
      "minHeight": 48,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
    },
    "medium": Object {
      "minHeight": 40,
      "paddingBottom": 12,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 12,
    },
    "small": Object {
      "minHeight": 32,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
    },
  },
<<<<<<< HEAD
=======
<<<<<<< HEAD
>>>>>>> origin/update/react
  "colors": Object {
    "natura": Object {
      "complementary": Object {
        "colorBrdNatCpBlue1": "#227BBD",
        "colorBrdNatCpBlue2": "#6BB1E2",
        "colorBrdNatCpBlue3": "#54BECF",
        "colorBrdNatCpBlue4": "#8E96A9",
        "colorBrdNatCpBrown1": "#371E14",
        "colorBrdNatCpBrown2": "#5D3B2A",
        "colorBrdNatCpBrown3": "#7A573C",
        "colorBrdNatCpGreen1": "#569A32",
        "colorBrdNatCpGreen2": "#8EBE2F",
        "colorBrdNatCpGreen3": "#C2CD00",
        "colorBrdNatCpGreen4": "#BEC950",
        "colorBrdNatCpOrange1": "#E8530E",
        "colorBrdNatCpOrange2": "#EE7D00",
        "colorBrdNatCpOrange3": "#F49B00",
        "colorBrdNatCpOrange4": "#FCC433",
        "colorBrdNatCpRed1": "#DC0719",
        "colorBrdNatCpRed2": "#E74627",
        "colorBrdNatCpRed3": "#ED743E",
        "colorBrdNatCpRed4": "#F29669",
        "colorBrdNatCpYellow1": "#CCB619",
        "colorBrdNatCpYellow2": "#E7CE22",
        "colorBrdNatCpYellow3": "#FFD600",
        "colorBrdNatCpYellow4": "#8E96A9",
      },
      "grayscale": Object {
        "colorBrdBlack": "#000000",
        "colorBrdNatGray": "#333333",
        "colorBrdNatGray1": "#777777",
        "colorBrdNatGray2": "#BBBBBB",
        "colorBrdNatGray3": "#EEEEEE",
      },
      "level": Object {
        "colorLevelBronze": "#C47F5B",
        "colorLevelDiamond": "#AEC2CE",
        "colorLevelGold": "#D8B75E",
        "colorLevelSeed": "#7B6A58",
        "colorLevelSilver": "#9D9D9C",
      },
      "primary": Object {
        "colorBrdNatPrimaryBrown": "#513528",
        "colorBrdNatPrimaryOrange": "#FF6B0B",
        "colorBrdNatPrimaryWhite": "#FFFFFF",
        "colorBrdNatPrimaryYellow": "#F4AB34",
      },
      "secondary": Object {
        "colorBrdNatSecondaryBlue": "#71CACD",
        "colorBrdNatSecondaryGreen": "#B3CA36",
        "colorBrdNatSecondaryPurple": "#7080A0",
        "colorBrdNatSecondaryYellow": "#FFBA1B",
      },
    },
  },
<<<<<<< HEAD
=======
=======
>>>>>>> origin/update/react
>>>>>>> origin/update/react
  "elevation": Object {
    "0": "none",
    "1": "0 1px 1px 0 rgba(0, 0, 0, 0.14)",
    "12": "0 12px 17px 0 rgba(0, 0, 0, 0.14)",
    "16": "0 16px 24px 0 rgba(0, 0, 0, 0.14)",
    "2": "0 2px 2px 0 rgba(0, 0, 0, 0.14)",
    "24": "0 24px 38px 0 rgba(0, 0, 0, 0.14)",
    "3": "0 3px 4px 0 rgba(0, 0, 0, 0.14)",
    "4": "0 4px 5px 0 rgba(0, 0, 0, 0.14)",
    "6": "0 6px 10px 0 rgba(0, 0, 0, 0.14)",
    "8": "0 8px 10px 0 rgba(0, 0, 0, 0.14)",
    "9": "0 9px 12px 0 rgba(0, 0, 0, 0.14)",
  },
  "fontSize": Object {
    "body1": Object {
      "fontSize": 16,
      "fontWeight": 400,
      "lineHeight": 24,
    },
    "body2": Object {
      "fontSize": 14,
      "fontWeight": 400,
      "lineHeight": 21,
    },
    "button": Object {
      "fontSize": 14,
      "fontWeight": 500,
      "lineHeight": 21,
    },
    "caption": Object {
      "fontSize": 12,
      "fontWeight": 400,
      "lineHeight": 18,
    },
    "h1": Object {
      "fontSize": 96,
      "fontWeight": 400,
      "lineHeight": 144,
    },
    "h2": Object {
      "fontSize": 60,
      "fontWeight": 400,
      "lineHeight": 90,
    },
    "h3": Object {
      "fontSize": 48,
      "fontWeight": 400,
      "lineHeight": 72,
    },
    "h4": Object {
      "fontSize": 34,
      "fontWeight": 400,
      "lineHeight": 51,
    },
    "h5": Object {
      "fontSize": 24,
      "fontWeight": 400,
      "lineHeight": 36,
    },
    "h6": Object {
      "fontSize": 20,
      "fontWeight": 500,
      "lineHeight": 30,
    },
    "overline": Object {
      "fontSize": 12,
      "fontWeight": 500,
      "lineHeight": 18,
    },
    "subtitle1": Object {
      "fontSize": 16,
      "fontWeight": 500,
      "lineHeight": 24,
    },
    "subtitle2": Object {
      "fontSize": 14,
      "fontWeight": 500,
      "lineHeight": 21,
    },
  },
  "iconSizes": Object {
    "huge": 144,
    "hugex": 192,
    "hugexx": 256,
    "large": 72,
    "largex": 96,
    "largexx": 128,
    "medium": 64,
    "micro": 16,
    "small": 32,
    "standard": 48,
    "tiny": 24,
  },
  "opacity": Object {
    "00": 0,
    "04": 0.04,
    "08": 0.08,
    "12": 0.12,
    "16": 0.16,
    "24": 0.24,
    "32": 0.32,
    "48": 0.48,
    "56": 0.56,
    "64": 0.64,
    "80": 0.8,
  },
  "overlay": Object {
    "solidBlack": 0.56,
  },
  "radius": Object {
    "circle": "50%",
    "large": 8,
    "medium": 4,
    "none": 0,
    "small": 2,
  },
  "spacing": Object {
    "spacing": 8,
    "spacingHuge": 64,
    "spacingLarge": 48,
    "spacingMicro": 4,
    "spacingNone": 0,
    "spacingSemi": 32,
    "spacingSmall": 16,
    "spacingStandard": 24,
    "spacingTiny": 8,
  },
}
`;
